wig/smooth.go

```diff
@@ -0,0 +1,28 @@
+package wig
+
+// Smooth performs moving average smoothing on an input Wig struct using a user-specified windowSize.
+func Smooth(w Wig, windowSize int) Wig {
+	var j, k, midPoint int
+	var sum float64
+	//first we make a fresh copy of the input wig to serve as our output.
+	var answer = Wig{StepType: "fixedStep", Chrom: w.Chrom, Start: w.Start, Step: w.Step, Span: w.Span, Values: make([]float64, len(w.Values))}
+	copy(answer.Values, w.Values)
+	for j = 0; j < len(w.Values)-windowSize; j++ {
+		sum = 0.0
+		for k = j; k < j+windowSize; k++ {
+			sum += w.Values[k]
+		}
+		midPoint = (j + j + windowSize) / 2
+		answer.Values[midPoint] = sum / float64(windowSize)
+	}
+	return answer
+}
+
+// SmoothSlice performs moving average smoothing on an input slice of Wig structs using a user-specified windowSize.
+func SmoothSlice(w []Wig, windowSize int) []Wig {
```

Comment:
I want to check that I understand Smooth and SmoothSlice. In cmd/wigMath, the input wig file becomes "records []Wig". SmoothSlice takes each wig in the []Wig, which is the wig block with a header line, and calls Smooth to assign the average wig value within a windowSize to the midpoint position of each window. Is this right?

Originally I was going to ask "Why is the midPoint = (j + j+ windowSize)/2, as opposed to something like midPoint=j+windowSize/2?" But then I realized that those 2 expressions are the same

Reply:
Now there will be a change in the program about missing data.
-For operations like add/multiplication, missing data in an input wig will cause output to be missing data
-For smoothing, missing data will also cause output to be missing data, as opposed to a smoothed over average where answer.Values[midPoint] = sum / float64(windowSize-numberOfPositionsWithMissingData)